packages/js/product-editor/src/components/attribute-combobox-field/types.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import type { ProductAttribute } from '@woocommerce/data';
+
+/*
+ * Define the attributes combobox control item type,
+ * which is a combination of the product attribute and
+ * additional properties.
+ */
+export type AttributesComboboxControlItem = ProductAttribute & {
+	isDisabled?: boolean;
+	takenBy?: number;
```

Comment:
What is `takenBy`? It's not yet used, right? I couldn't figure out just by the name.